services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -88,12 +98,62 @@ async def collect_garbage(registry: RedisResourceRegistry, app: web.Application)
                     app=app,
                     project_uuid=resource_value,
                     user_id=int(dead_key["user_id"]),
+                    disable_access_check=False,
                 )
 
+    # try to remove users which were marked as GUESTS manually
+    await remove_users_manually_marked_as_guests(
+        app=app, alive_keys=alive_keys, dead_keys=dead_keys
+    )
+
     # remove possible pending contianers
     await remove_orphaned_services(registry, app)
 
 
+async def remove_users_manually_marked_as_guests(
+    app: web.Application,
+    alive_keys: List[Dict[str, str]],
+    dead_keys: List[Dict[str, str]],
+) -> None:
+    """
+    Removes all the projects associated with GUEST users in the system.
+    If the user defined a TEMPLATE, this one also gets removed.
+    """
+
+    user_ids_to_ignore = set()
+    for entry in chain(alive_keys, dead_keys):
```

Comment:
why do you chain the dead keys here? aren't they meant to be dead? so ok to remove?

Reply:
Because there are 2 types of GUEST users, the ones generated from opening a link and the ones added manually.
The ones marked as guests are treated differently from the ones detected via websocket disconnection.